jsapp/js/components/formEditors.es6

```diff
@@ -37,330 +39,766 @@ import {
+      IMPORT_URL: 'import-url',
+      PROJECT_DETAILS: 'project-details'
+    }
+
+    const formAsset = this.props.formAsset;
+
+    this.state = {
+      isSessionLoaded: !!session.currentAccount,
+      isSubmitPending: false,
+      formAsset: formAsset,
+      // project details
+      name: formAsset ? formAsset.name : '',
+      description: formAsset ? formAsset.settings.description : '',
+      sector: formAsset ? formAsset.settings.sector : null,
+      country: formAsset ? formAsset.settings.country : null,
+      'share-metadata': formAsset ? formAsset.settings['share-metadata'] : false,
+      // steps
+      currentStep: null,
+      previousStep: null,
+      // importing url
+      isImportFromURLPending: false,
       importUrl: '',
       importUrlButtonEnabled: false,
-      importUrlButton: t('Import')
-    }
-    if (this.props.initialData !== undefined) {
-      assign(state, this.props.initialData);
-    }
-    this.state = state;
+      importUrlButton: t('Import'),
+      // template
+      isApplyTemplatePending: false,
+      applyTemplateButton: t('Next'),
+      chosenTemplateUid: null,
+      // upload files
+      isUploadFilePending: false
+    };
+
     autoBind(this);
   }
-  componentDidMount () {
+
+  /*
+   * setup
+   */
+
+  componentDidMount() {
+    this.setInitialStep();
     this.listenTo(session, (session) => {
       this.setState({
-        sessionLoaded: true,
+        isSessionLoaded: true,
       });
     });
   }
-  nameChange (evt) {
-    this.setState({
-      name: evt.target.value
-    });
+
+  setInitialStep() {
+    switch (this.props.context) {
+      case PROJECT_SETTINGS_CONTEXTS.NEW:
+      case PROJECT_SETTINGS_CONTEXTS.REPLACE:
+        return this.displayStep(this.STEPS.FORM_SOURCE);
+      case PROJECT_SETTINGS_CONTEXTS.EXISTING:
+      case PROJECT_SETTINGS_CONTEXTS.BUILDER:
+        return this.displayStep(this.STEPS.PROJECT_DETAILS);
+      default:
+        throw new Error(`Unknown context: ${this.props.context}!`);
+    }
   }
-  descriptionChange (evt) {
-    this.setState({
-      description: evt.target.value
-    });
+
+  getBaseTitle() {
+    switch (this.props.context) {
+      case PROJECT_SETTINGS_CONTEXTS.NEW:
+        return t('Create project');
+      case PROJECT_SETTINGS_CONTEXTS.REPLACE:
+        return t('Replace project');
+      case PROJECT_SETTINGS_CONTEXTS.EXISTING:
+      case PROJECT_SETTINGS_CONTEXTS.BUILDER:
+      default:
+        return t('Project settings');
+    }
   }
-  countryChange (val) {
+
+  getStepTitle(step) {
+    switch (step) {
+      case this.STEPS.FORM_SOURCE: return t('Choose a source');
+      case this.STEPS.CHOOSE_TEMPLATE: return t('Choose template');
+      case this.STEPS.UPLOAD_FILE: return t('Upload XLSForm');
+      case this.STEPS.IMPORT_URL: return t('Import XLSForm');
+      case this.STEPS.PROJECT_DETAILS: return t('Project details');
+      default: return '';
+    }
+  }
+
+  getFilenameFromURI(url) {
```

Comment:
@magicznyleszek, it looks like this was introduced in https://github.com/kobotoolbox/kpi/commit/e32e706d0db1680ef3085fef93cf36c5d403df14 but isn't used anymore as of https://github.com/kobotoolbox/kpi/commit/33ffa8bea60efb4d0a723286403a7751a4ea06d4